stmhal/mphalport.h

```diff
@@ -6,10 +6,16 @@
 // go in some MCU-specific header, but for now it lives here.
 #if defined(MCU_SERIES_F4)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1fff7a10)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size)
+#define MP_HAL_CLEANCACHE(addr, size)
 #elif defined(MCU_SERIES_F7)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1ff0f420)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size) (SCB_CleanInvalidateDCache_by_Addr((uint32_t*)(((uint32_t)addr ) & ~(0x1F)), (size) + ((uint32_t)addr - (((uint32_t)addr ) & ~(0x1F)))))
```

Comment:
I omitted the casts for clarity.

Here's an example where I think that the original code doesn't work.

Let's suppose an address of 0x1008 and a size of 0x20. This covers 0x1008 thru 0x1027 which covers 2 cache lines. 0x1000-0x101f and 0x1020-0x103f

addr & ~0x1f takes the address down to 0x1000.

size + (addr - (addr & ~0x1F)) becomes 0x20 + (0x1008 - 0x1000) = 0x28

What you want is to transform 0x1008 for a length of 0x20 to become 0x1000 for a length of 0x40 (so the address is cache-line aligned, and the length is cache line aligned).